src/transformers/safetensors_conversion.py

```diff
@@ -0,0 +1,105 @@
+from typing import Optional
+
+from huggingface_hub import Discussion, HfApi
+
+from .utils import cached_file, logging
+
+
+logger = logging.get_logger(__name__)
+
+
+def previous_pr(api: "HfApi", model_id: str, pr_title: str) -> Optional["Discussion"]:
+    try:
+        main_commit = api.list_repo_commits(model_id)[0].commit_id
+        discussions = api.get_repo_discussions(repo_id=model_id)
+    except Exception as e:
+        logger.info(f"Could not retrieve repository discussions: {repr(e)}")
+        return None
+    for discussion in discussions:
+        if discussion.status == "open" and discussion.is_pull_request and discussion.title == pr_title:
+            commits = api.list_repo_commits(model_id, revision=discussion.git_reference)
+
+            if main_commit == commits[1].commit_id:
+                return discussion
+    return None
+
+
+def spawn_conversion(token: str, private: bool, model_id: str):
+    logger.info("Attempting to convert .bin model on the fly to safetensors.")
+
+    try:
+        import json
+        import uuid
+
+        import requests
+    except (ImportError, ModuleNotFoundError) as e:
+        raise ValueError("Could not perform on-the-fly conversion as `requests` isn't installed.") from e
+
+    sse_url = "https://safetensors-convert.hf.space/queue/join"
+    sse_data_url = "https://safetensors-convert.hf.space/queue/data"
```

Comment:
I would add a constant variable `SAFETENSORS_CONVERT_SPACE_URL = https://safetensors-convert.hf.space"` just in case (not necessarily an environment variable though)